### ISSUE TITLE
uninitialized constant Facter::Util::FileRead on CentOS 7 with puppetlabs-pc1 (Puppet 4)

### ISSUE THREAD

**Issue by olifre** [treydock/puppet-infiniband#6]
Using this module, without even instantiating the class, I get:
```
Error: Facter: error while resolving custom facts in /opt/puppetlabs/puppet/cache/lib/facter/infiniband_board_id.rb: uninitialized constant Facter::Util::FileRead
Error: Facter: error while resolving custom facts in /opt/puppetlabs/puppet/cache/lib/facter/infiniband_fw_version.rb: uninitialized constant Facter::Util::FileRead
Error: Facter: error while resolving custom facts in /opt/puppetlabs/puppet/cache/lib/facter/infiniband_rate.rb: uninitialized constant Facter::Util::FileRead
```
Instantiating the class, I get additionally:
```
Error: Could not retrieve catalog from remote server: Error 500 on SERVER: Server Error: Evaluation Error: Error while evaluating a Resource Statement, Evaluation Error: Resource type not found: Shellvar at /etc/puppetlabs/code/environments/production/modules/infiniband/manifests/config.pp:5:3 on node XXXXX
```
Is this a bug? Am I missing any special dependencies?

**Comment by olifre** [comment]
While I guess the second issue could be resolved by adding https://github.com/hercules-team/augeasproviders_shellvar , I don't know what to do about the Facter problem...

**Comment by treydock** [comment]
should be resolved with #7
